core/time/shared/now.rb

```diff
@@ -5,4 +5,21 @@
     TimeSpecs::SubTime.send(@method).should be_an_instance_of(TimeSpecs::SubTime)
     TimeSpecs::MethodHolder.send(@method).should be_an_instance_of(Time)
   end
+
+  it "sets the current time" do
+    check = `date "+%H:%M:%S"`.strip
+    time = check.match(/(\d{2})\:(\d{2})\:(\d{2})/)
+    now = TimeSpecs::MethodHolder.send(@method)
+
+    now.hour.should == time[1].to_i
+    now.min.should == time[2].to_i
+    now.sec.should == time[3].to_i
```

Comment:
Unfortunately some time could pass between the check above and this expectation.
This applies to all units, because if it's 23:59:59.999, we only need 1ms in between to fail the spec.
Also `date` doesn't work on Windows.

We could maybe use `Process.clock_gettime` to get an estimation, assuming that one is correct:
```ruby
Time.now.to_f.should be_close(Process.clock_gettime(Process::CLOCK_REALTIME), 10.0)
```

Reply:
Date.today and DateTime.now could then be converted to a Time using their year, month, etc values and compared to Time.now.